osquery/tables/system/windows/processes.cpp

```diff
@@ -147,238 +214,334 @@ Status getProcList(std::set<long>& pids) {
-    /// Get the process UID and GID from its SID
-    HANDLE tok = nullptr;
-    std::vector<char> tokUser(sizeof(TOKEN_USER), 0x0);
-    auto ret = OpenProcessToken(hProcess, TOKEN_READ, &tok);
-    if (ret != 0 && tok != nullptr) {
-      unsigned long tokOwnerBuffLen;
-      ret = GetTokenInformation(tok, TokenUser, nullptr, 0, &tokOwnerBuffLen);
-      if (ret == 0 && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
-        tokUser.resize(tokOwnerBuffLen);
-        ret = GetTokenInformation(
-            tok, TokenUser, tokUser.data(), tokOwnerBuffLen, &tokOwnerBuffLen);
-      }
+  unsigned long size_out = 0;
+  PROCESS_BASIC_INFORMATION proc_info;
+  SecureZeroMemory(&proc_info, sizeof(PROCESS_BASIC_INFORMATION));
 
-      // Check if the process is using an elevated token
-      auto elevated = FALSE;
-      TOKEN_ELEVATION Elevation;
-      DWORD cbSize = sizeof(TOKEN_ELEVATION);
-      if (GetTokenInformation(
-              tok, TokenElevation, &Elevation, sizeof(Elevation), &cbSize)) {
-        elevated = Elevation.TokenIsElevated;
-      }
+  // See the Process Hacker implementation for more details on the hard coded 60
+  // https://github.com/processhacker/processhacker/blob/master/phnt/include/ntpsapi.h#L160
+  auto ret = kNtQueryInformationProcess(proc, 60, NULL, 0, &size_out);
 
-      r["is_elevated_token"] = elevated ? INTEGER(1) : INTEGER(0);
-    }
-    if (uid != 0 && ret != 0 && !tokUser.empty()) {
-      auto sid = PTOKEN_OWNER(tokUser.data())->Owner;
-      r["uid"] = INTEGER(getUidFromSid(sid));
-      r["gid"] = INTEGER(getGidFromSid(sid));
-    } else {
-      r["uid"] = INTEGER(uid);
-      r["gid"] = INTEGER(gid);
-    }
-    if (tok != nullptr) {
-      CloseHandle(tok);
-      tok = nullptr;
-    }
+  if (ret != STATUS_BUFFER_OVERFLOW && ret != STATUS_BUFFER_TOO_SMALL &&
+      ret != STATUS_INFO_LENGTH_MISMATCH) {
+    VLOG(1) << "NtQueryInformationProcess failed for " << pid << " with "
+            << ret;
+    return;
   }
 
-  if (hProcess != nullptr) {
-    CloseHandle(hProcess);
+  PVOID cmdline = static_cast<PVOID>(malloc(size_out));
+  ret = kNtQueryInformationProcess(proc, 60, cmdline, size_out, &size_out);
+  if (!NT_SUCCESS(ret)) {
+    VLOG(1) << "NtQueryInformationProcess failed for " << pid << " with "
+            << ret;
+    return;
   }
+  auto ustr = reinterpret_cast<PUNICODE_STRING>(cmdline);
+  out = wstringToString(ustr->Buffer);
 }
 
-// collect perf data into a hashmap by pid to later be refferenced
+void getProcessPathInfo(HANDLE& proc, unsigned long pid, Row& r) {
+  unsigned long out = kMaxPathSize;
+  std::shared_ptr<char> path(new char[kMaxPathSize],
+                             std::default_delete<char[]>());
+  SecureZeroMemory(path.get(), kMaxPathSize);
+  auto ret = QueryFullProcessImageName(proc, 0, path.get(), &out);
+  if (ret != TRUE) {
+    VLOG(1) << "Failed to lookup path information for process " << pid;
+    r["path"] = "-1";
+  } else {
+    r["path"] = TEXT(path.get());
```

Comment:
@akindyakov Does this support std::move semantics and is this usage here generally worth it?